scripts/test/Muon/elemental_analysis/elemental_analysis_test.py

```diff
@@ -0,0 +1,502 @@
+        mock_warning.assert_called_with(warning_text)
+
+    def test_gamms_checked_calls_checked_data_for_each_element(self):
+        # TODO remove -2 on element number once json file has been restructured
+        elem = len(self.gui.ptable.peak_data)-2
+        self.gui.checked_data = mock.Mock()
+        self.gui.gammas_checked()
+        self.assertEqual(self.gui.checked_data.call_count, elem)
+
+    def test_gamms_unchecked_calls_checked_data_for_each_element(self):
+        # TODO remove -2 on element number once json file has been restructured
+        elem = len(self.gui.ptable.peak_data)-2
+        self.gui.checked_data = mock.Mock()
+        self.gui.gammas_unchecked()
+        self.assertEqual(self.gui.checked_data.call_count, elem)
+
+    def test_major_checked_calls_checked_data_for_each_element(self):
+        # TODO remove -2 on element number once json file has been restructured
+        elem = len(self.gui.ptable.peak_data)-2
+        self.gui.checked_data = mock.Mock()
+        self.gui.major_peaks_checked()
+        self.assertEqual(self.gui.checked_data.call_count, elem)
+
+    def test_major_unchecked_calls_checked_data_for_each_element(self):
+        # TODO remove -2 on element number once json file has been restructured
+        elem = len(self.gui.ptable.peak_data)-2
+        self.gui.checked_data = mock.Mock()
+        self.gui.major_peaks_unchecked()
+        self.assertEqual(self.gui.checked_data.call_count, elem)
+
+    def test_minor_checked_calls_checked_data_for_each_element(self):
+        # TODO remove -2 on element number once json file has been restructured
+        elem = len(self.gui.ptable.peak_data)-2
+        self.gui.checked_data = mock.Mock()
+        self.gui.minor_peaks_checked()
+        self.assertEqual(self.gui.checked_data.call_count, elem)
+
+    def test_minor_unchecked_calls_checked_data_for_each_element(self):
+        # TODO remove -2 on element number once json file has been restructured
+        elem = len(self.gui.ptable.peak_data)-2
+        self.gui.checked_data = mock.Mock()
+        self.gui.minor_peaks_unchecked()
+        self.assertEqual(self.gui.checked_data.call_count, elem)
+
+    @mock.patch('Muon.GUI.ElementalAnalysis.elemental_analysis.ElementalAnalysisGui._update_peak_data')
+    def test_checked_data_changes_all_states_in_list(self, mock_update_peak_data):
+        selection = [mock.Mock() for i in range(10)]
+        self.gui.checked_data('Cu', selection, True)
+
+        self.assertTrue(all(map(lambda m: m.setChecked.call_count == 1, selection)))
+        mock_update_peak_data.assert_called_with('Cu')
+
+    def test_get_electron_peaks_returns_a_dict_with_correct_length(self):
+        peaks = len(self.gui.ptable.peak_data["Electrons"])
+        electron_dict = self.gui._get_electron_peaks()
+        self.assertEqual(len(electron_dict), peaks)
+        for _, peak in electron_dict.items():
+            float(peak)
```

Comment:
a few tests will need updating once #26444 is merged

Reply:
Added a reminder for it.

Reply:
if you merge master the file changes should be in now